hydra-node/test/Hydra/NodeSpec.hs

```diff
@@ -0,0 +1,270 @@
+            , signed = UnsafeSigned (SigMockDSIGN "c1a00d60aabde67a" 20)
+            , Msg.snapshotNumber = 2
+            }
+      }
+  , NetworkEvent
+      { message =
+          ReqTx
+            { Msg.party = 10
+            , transaction = SimpleTx{txSimpleId = 5, txInputs = utxoRefs [12, 15, 22], txOutputs = utxoRefs [26, 27, 28]}
+            }
+      }
+  , NetworkEvent
+      { message =
+          ReqTx
+            { Msg.party = 10
+            , transaction = SimpleTx{txSimpleId = 6, txInputs = utxoRefs [16, 19, 23, 24, 25], txOutputs = utxoRefs [29]}
+            }
+      }
+  , NetworkEvent
+      { message =
+          ReqTx
+            { Msg.party = 10
+            , transaction = SimpleTx{txSimpleId = 7, txInputs = utxoRefs [26, 28, 29], txOutputs = utxoRefs [30, 31, 32, 33, 34, 35, 36]}
+            }
+      }
+  , NetworkEvent
+      { message =
+          ReqTx
+            { Msg.party = 10
+            , transaction = SimpleTx{txSimpleId = 8, txInputs = utxoRefs [30, 33, 34], txOutputs = utxoRefs [37, 38]}
+            }
+      }
+  , NetworkEvent
+      { message =
+          ReqTx
+            { Msg.party = 10
+            , transaction = SimpleTx{txSimpleId = 9, txInputs = utxoRefs [27, 32, 36, 37, 38], txOutputs = utxoRefs [39, 40, 41]}
+            }
+      }
+  , NetworkEvent
+      { message =
+          ReqTx
+            { Msg.party = 10
+            , transaction = SimpleTx{txSimpleId = 10, txInputs = utxoRefs [31, 35], txOutputs = utxoRefs [42, 43, 44, 45, 46, 47]}
+            }
+      }
+  , NetworkEvent
+      { message =
+          ReqSn
+            { Msg.party = 10
+            , Msg.snapshotNumber = 3
+            , transactions =
+                [ SimpleTx
+                    { txSimpleId = 5
+                    , txInputs = utxoRefs [12, 15, 22]
+                    , txOutputs = utxoRefs [26, 27, 28]
+                    }
+                , SimpleTx{txSimpleId = 6, txInputs = utxoRefs [16, 19, 23, 24, 25], txOutputs = utxoRefs [29]}
+                , SimpleTx{txSimpleId = 7, txInputs = utxoRefs [26, 28, 29], txOutputs = utxoRefs [30, 31, 32, 33, 34, 35, 36]}
+                , SimpleTx{txSimpleId = 8, txInputs = utxoRefs [30, 33, 34], txOutputs = utxoRefs [37, 38]}
+                ]
+            }
+      }
+  , NetworkEvent
+      { message =
+          AckSn
+            { Msg.party =
+                30
+            , signed = UnsafeSigned (SigMockDSIGN "7d81c3daff6fe561" 30)
+            , Msg.snapshotNumber = 3
+            }
+      }
+  ]
+
+-- | "Prepare" a node by feeding it a list of 'Event's.
+-- Throws a 'FeedError' exception containig a list of 'LogicError'
+-- if there are any errors in the processing of events.
+feedEvents ::
+  ( Tx tx
+  , MonadSTM m
+  , MonadThrow (STM m)
+  ) =>
+  HydraNode tx m ->
+  [Event tx] ->
+  STM m (HydraNode tx m)
+feedEvents node eventsList = do
+  (errs, _) <- partitionEithers <$> traverse (processNextEvent node) eventsList
```

Comment:
Better to have this function here, but I still do not think this is the right level of "hooking into the logic" given a list of events. Or it doesn't "feel right" at least:

`processNextEvent` is producing effects which we do not treat here and the only side effect we want to have is managing the `HeadState`. So far so good.

However, given a list of `Event` this might not behave exactly as in "in vivo" as nothing would handle `Delay` effects by re-enqueuing them. Which brings me to the question "Why use `processNextEvent` and not directly `Logic.update` for replaying a recorded, potentially re-enqueued list of events?" That function would be more explicit about the `Wait` outcome and `feedEvents` here would need to make a "more concious" decision about those outcomes.